sentry-ruby/spec/sentry_spec.rb

```diff
@@ -208,6 +208,107 @@
   end
 
   describe ".start_transaction" do
+    describe "sampler example" do
+      before do
+        perform_basic_setup do |config|
+          config.traces_sampler = lambda do |sampling_context|
+            # if this is the continuation of a trace, just use that decision (rate controlled by the caller)
+            unless sampling_context[:parent_sampled].nil?
+              next sampling_context[:parent_sampled]
+            end
+
+            # transaction_context is the transaction object in hash form
+            # keep in mind that sampling happens right after the transaction is initialized
+            # e.g. at the beginning of the request
+            transaction_context = sampling_context[:transaction_context]
+
+            # transaction_context helps you sample transactions with more sophistication
+            # for example, you can provide different sample rates based on the operation or name
+            op = transaction_context[:op]
+            transaction_name = transaction_context[:name]
+
+            case op
+            when /request/
+              case transaction_name
+              when /health_check/
+                0.0
+              when /payment/
+                0.5
+              when /api/
+                0.2
+              else
+                0.1
+              end
+            when /sidekiq/
+              0.01 # you may want to set a lower rate for background jobs if the number is large
+            else
+              0.0
+            end
+          end
+        end
+      end
+
+      it "prioritizes parent's sampling decision" do
+        sampled_trace = "d298e6b033f84659928a2267c3879aaa-2a35b8e9a1b974f4-1"
+        unsampled_trace = "d298e6b033f84659928a2267c3879aaa-2a35b8e9a1b974f4-0"
+
+        parent_transaction = Sentry::Transaction.from_sentry_trace(sampled_trace)
+        transaction = described_class.start_transaction(transaction: parent_transaction, op: "rack.request", name: "/payment", hub: Sentry.get_current_hub)
```

Comment:
```suggestion
       transaction = Sentry::Transaction.from_sentry_trace(sampled_trace)
       described_class.start_transaction(transaction: parent_transaction, op: "rack.request", name: "/payment", hub: Sentry.get_current_hub)
```

Since this isn't actually the parent transaction (that's in another service) and since `start_transaction` is just a mutator when given an already-existing transaction.